packages/search/src/content/index.ts

```diff
@@ -89,13 +89,195 @@ export class ContentSearchService
+ * //////////////////////////////////////////////////////////
+ * // Search for all private content associated with a site
+ * ///////////////////////////////////////////////////////////
+ *
+ * // TODO: As of 8/17/21 the hub api has a bug in which
+ * // Any query with a not filter on the access field
+ * // (e.g access: not('public')) will return no results
+ *
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    // Note: only private content visible to the authenticated user will be returned
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    access: {
+ *        value: ['public'],
+ *        bool: IBooleanOperator.NOT
+ *    }
+ * }
+ * const searchResults = await searchContent({ filter, options });
+ *
+ * ...
+ * /////////////////////////////////
+ * // Search for a specific item
+ * /////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    portal: 'https://www.arcgis.com',
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    id: 'my_item_id' // Can also be expressed as { value: ['my_item_id'] }
+ * }
+ * const searchResults = await searchContent({ filter, options });
  * ```
+ * There are a couple gotchas that need to be accounted for:
+ *
+ * 1) There is no way to specify the number of results per page. If you need to fetch all
+ * items that match a given query, you'll need to utilize the `hasNext` flag as well as the
+ * `next` function included on the return object and make multiple XHR requests.
+ *
+ * 2) There are differences in item structure for results returned from the
+ * hub index vs AGO/Enterprise (i.e. 'public' items vs 'private' items).
+ * Here is an example of a 'private' map result:
+ * ```js
+ * {
+ *  access: "myself",
+ *  appCategories: [],
+ *  avgRating: 0,
+ *  categories: [],
+ *  collection: ["Map"],
+ *  contentOrigin: "self",
+ *  created: 1623945553000,
+ *  culture: "en-us",
+ *  description: “A description",
+ *  extent: {
+ *    coordinates: [],
+ *    type: "envelope"
+ *  },
+ *  hubType: "map",
+ *  id: "06c0cdadc2ec48509576c20da8572bf8",
+ *  industries: [],
+ *  isOrgItem: true,
+ *  languages: [],
+ *  listed: false,
+ *  modified: 1629220743000,
+ *  name: "Traffic Camera Enforcement Locations in Washington DC",
+ *  numComments: 0,
+ *  numRatings: 0,
+ *  numViews: 0,
+ *  owner: "juliana_pa",
+ *  properties: {...}
+ *  protected: false,
+ *  scoreCompleteness: 66,
+ *  screenshots: [],
+ *  searchDescription: “A description”,
+ *  size: -1,
+ *  subInfo: 0,
+ *  tags: ["DC GIS", "police", "speed", "speeding", "ticket", "traffic", "violation"],
+ *  thumbnail: "thumbnail/ago_downloaded.png",
+ *  title: "Traffic Camera Enforcement Locations in Washington DC",
+ *  type: "Web Map",
+ *  typeKeywords: ["ArcGIS Online", "Explorer Web Map", "Map", "Online Map", "Web Map"],
+ * }
+ * ```
```

Comment:
For some reason I get syntax highlighting on the next snippet, but not this one. Not a huge deal.

Reply:
Really? I thought the syntax highlighting worked on both for me but I'll double check